app/models/spree/gateway/adyen_credit_card.rb

```diff
@@ -36,20 +36,36 @@ def authorize(amount, card, gateway_options)
       ActiveMerchant::Billing::Response.new(true, "dummy authorization response")
     end
 
+    def perform_authorization_3d(payment, redirect_response_params)
+      response = rest_client.authorise_payment_3dsecure(authorization_3d_request(payment, redirect_response_params))
+      unless response.success?
```

Comment:
Everythinng but the first line here is duplicating what `authorise_new_payment` already does. Could we change `authorize_new_payment` to handle both and call out to something to perform the 3d secure auth if necessary?